csharp/src/Ice/Internal/Instance.cs

```diff
@@ -580,7 +580,15 @@ internal void initialize(Ice.Communicator communicator, Ice.InitializationData i
 
         try
         {
-            _initData.properties ??= new Ice.Properties();
+            _initData.properties ??= new Properties();
```

Comment:
In C++ version this assignment is done with the mutex locked, should we lock the mutex here too?

Reply:
It doesn't matter. This initData is not supposed to be shared by multiple communicators, and initData itself is not thread-safe.